app/components/atoms/LanguageProvider/reducer.js

```diff
@@ -22,12 +22,11 @@ export const initialState = fromJS({
 /* eslint-disable default-case, no-param-reassign */
 export const languageProviderReducer = (state = initialState, action) =>
   produce(state, (/* draft */) => {
-    switch (action.type) {
-      case languageProviderTypes.CHANGE_LOCALE:
-        return state.set('locale', action.locale);
-      default:
-        return state;
-    }
+    const actionType = {
+      [languageProviderTypes.CHANGE_LOCALE]: () =>
+        state.set('locale', action.locale)
+    };
+    return action.type in actionType ? actionType[action.type]() : state;
```

Comment:
this would be more expensive than the switch case since we're doing one iteration to create the actionType object and another to look through it